Compile error on gcc 11 (Fedora 34)

I'm on Fedora 33 x64, doing a Fedora 34 x64 mock build for john  ```1.9.0-Jumbo-1-464-gc9825e688```.

F34 has gcc 11:
```
# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/11/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.0.0 20210210 (Red Hat 11.0.0-0) (GCC) 
```

The rpm spec file uses these build commands:
```
cd src
./configure --disable-rexgen --enable-asan
make clean
make %{?_smp_mflags}
```

The error:
```
gcc -DAC_BUILT -mavx2 -DJOHN_AVX2 -c -g  -Og -fsanitize=address -m64 -I/usr/local/include -DARCH_LITTLE_ENDIAN=1 -DPREFER_FLOCK  -Wall -fno-omit-frame-pointer -Wno-deprecated-declarations -Wunused-but-set-variable -std=gnu89 -Wdate-time -D_POSIX_SOURCE -D_GNU_SOURCE -D_XOPEN_SOURCE=600  -fopenmp  -pthread -I/usr/local/include -DCL_SILENCE_DEPRECATION -funroll-loops argon2_encoding_plug.c -o argon2_encoding_plug.o
In file included from argon2_core_plug.c:32:
blake2.h:112:5: error: size of array element is not a multiple of its alignment
  112 |     blake2b_state S[4][1];
      |     ^~~~~~~~~~~~~
blake2.h:113:5: error: size of array element is not a multiple of its alignment
  113 |     blake2b_state R[1];
      |     ^~~~~~~~~~~~~
make[1]: *** [Makefile:1592: argon2_core_plug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/builddir/build/BUILD/john-1.9.0.jumbo.1/src'
make: *** [Makefile:190: default] Error 2
```

### Steps to reproduce ###
- Install a Fedora 33 machine.
- Install mock

- You can reproduce it by using the repo 
```
https://gitlab.com/fhuberts/rpmsUpstream.git
```
at commit ```315a54e```.

- This repo needs a checkout of
```
https://gitlab.com/fhuberts/rpmbuilder.git
```
alongside it.

- You have ```rpmbuilder``` and ```rpmsUpstream``` in the same directory.

- Next you do
```
cd ./rpmsUpstream/fedora/john
```

- And then
```
../buildRPMs  -M /etc/mock/fedora-34-x86_64.cfg rpm.spec
```

FWIW, I cannot build gcc 11-20210307 (and the previous snapshot) on the same system and using the same build script that built 10.1 fine, now failing like this:

```
config.status: creating config.h
config.status: executing default commands
make[2]: Leaving directory `/home/gcc/build/gcc-11-20210307/objdir'
make[1]: *** [stage1-bubble] Error 2
```

There's nothing obviously causing this error in the preceding messages. I'm not going to spend more time on this just yet. It could be some change between 10 and 11, or it could be a difference between releases and snapshots. There isn't a gcc 11 release yet.

> Is this still the case?

Yes
```
gcc --version
gcc (Ubuntu 20210225-1ubuntu1) 11.0.0 20210225 (experimental) [master revision 4028d01a050:8aeb92daee4:a6baafcac5308be1a5d92c0b2a179495b7a24b52]
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

```
In file included from argon2_core_plug.c:32:
blake2.h:112:5: error: size of array element is not a multiple of its alignment
  112 |     blake2b_state S[4][1];
      |     ^~~~~~~~~~~~~
blake2.h:113:5: error: size of array element is not a multiple of its alignment
  113 |     blake2b_state R[1];
      |     ^~~~~~~~~~~~~

```

--- a/src/blake2.h
+++ b/src/blake2.h
@@ -57,7 +57,7 @@ extern "C" {
     uint8_t  personal[BLAKE2S_PERSONALBYTES];  // 32
   } blake2s_param;
 
-  JTR_ALIGN( 64 ) typedef struct __blake2s_state
+  typedef struct JTR_ALIGN( 64 ) __blake2s_state
   {
     uint32_t h[8];
     uint32_t t[2];
@@ -82,7 +82,7 @@ extern "C" {
     uint8_t  personal[BLAKE2B_PERSONALBYTES];  // 64
   } blake2b_param;
 
-  JTR_ALIGN( 64 ) typedef struct __blake2b_state
+  typedef struct JTR_ALIGN( 64 ) __blake2b_state
   {
     uint64_t h[8];
     uint64_t t[2];
@@ -94,7 +94,7 @@ extern "C" {
 #if defined(JOHN_NO_SIMD) || (!defined(__SSE2__) && !defined(__SSE4_1__) && !defined(__XOP__))
   typedef struct __blake2sp_state
 #else
-  JTR_ALIGN( 64 ) typedef struct __blake2sp_state
+  typedef struct JTR_ALIGN( 64 ) __blake2sp_state
 #endif
   {
     blake2s_state S[8][1];
@@ -106,7 +106,7 @@ extern "C" {
 #if defined(JOHN_NO_SIMD) || (!defined(__SSE2__) && !defined(__SSE4_1__) && !defined(__XOP__))
   typedef struct __blake2bp_state
 #else
-  JTR_ALIGN( 64 ) typedef struct __blake2bp_state
+  typedef struct JTR_ALIGN( 64 ) __blake2bp_state
 #endif
   {
     blake2b_state S[4][1];

